src/middlewared/middlewared/api/v25_04_0/acme_protocol.py

```diff
@@ -0,0 +1,103 @@
+from pydantic import Field, Secret
+
+from middlewared.api.base import BaseModel, Excluded, excluded_field, single_argument_args
+
+
+__all__ = [
+    'ACMERegistrationCreateArgs', 'ACMERegistrationCreateResult', 'DNSAuthenticatorUpdateArgs',
+    'DNSAuthenticatorUpdateResult', 'DNSAuthenticatorCreateArgs', 'DNSAuthenticatorCreateResult',
+    'DNSAuthenticatorDeleteArgs', 'DNSAuthenticatorDeleteResult', 'DNSAuthenticatorSchemasArgs',
+    'DNSAuthenticatorSchemasResult', 'ACMERegistrationEntry', 'ACMEDNSAuthenticatorEntry',
+]
+
+
+class JWKCreate(BaseModel):
+    key_size: int = 2048
+    public_exponent: int = 65537
+
+
+class ACMERegistrationEntry(BaseModel):
+    id: int
+    uri: str
+    directory: str
+    tos: str
+    new_account_uri: str
+    new_nonce_uri: str
+    new_order_uri: str
+    revoke_cert_uri: str
+
+
+class ACMEDNSAuthenticatorEntry(BaseModel):
+    id: int
+    authenticator: str
+    attributes: Secret[dict]
+    name: str
+
+
+class DNSAuthenticatorCreate(ACMEDNSAuthenticatorEntry):
+    id: Excluded = excluded_field()
+
+
+class DNSAuthenticatorUpdate(DNSAuthenticatorCreate):
```

Comment:
You should include `metaclass=ForUpdateMetaclass` here (as we do for other update schemas) so that all fields become not required